service/collector.go

```diff
@@ -183,7 +183,9 @@ LOOP:
 			col.logger.Info("Received shutdown request")
 			break LOOP
 		case <-ctx.Done():
-			col.logger.Error("Context done, terminating process", zap.Error(ctx.Err()))
+			// Log the error for the context as a string or else zap.Error prints a stack trace.
+			col.logger.Info("Context done, terminating process", zap.String("Context Error", ctx.Err().Error()))
```

Comment:
Wrote a test to prove this. Is the "error" log level that prints the stacktrace, not the zap.Error field. So this PR can be as simple as changing `col.logger.Error` to `col.logger.Info`

Reply:
Good catch. Not sure how I got confused on it but appreciate the double check.